src/load_factions.rs

```diff
@@ -0,0 +1,234 @@
+// Most of this code was taken from the bevy_common_assets crate:
+// https://github.com/NiklasEi/bevy_common_assets/blob/main/src/json.rs
+// There are also examples of the same with JSON, RON, TOML and more if we want to switch later.
+
+use bevy::app::{App, Plugin};
+use bevy::asset::io::Reader;
+use bevy::asset::*;
+use bevy::prelude::*;
+use serde::Deserialize;
+use serde_json::from_slice;
+use thiserror::Error;
+
+/// Plugin to load your asset type [`FactionAsset`] from json files.
+pub struct FactionLoaderPlugin;
+
+impl Plugin for FactionLoaderPlugin {
+    fn build(&self, app: &mut App) {
+        app.register_asset_loader::<FactionAssetLoader>(FactionAssetLoader)
+            .init_asset::<FactionAsset>()
+            .insert_resource(Factions(vec![]))
+            .add_systems(Startup, load_factions)
+            .add_systems(
+                Update,
+                (
+                    setup_factions_resource.after(load_factions),
+                    display_content.after(setup_factions_resource), // TODO: Remove display_content, when actually using the factions resource.
```

Comment:
Are we accepting `TODO` as comments, or should we make them as issues for tracking?

Reply:
Der er oprettet opgave på den.